property-dump-plugin/src/main/java/org/sonar/dump/PropertyDumpPlugin.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * property-dump-plugin
+ * Copyright (C) 2009-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.dump;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import org.sonar.api.Plugin;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+
+public class PropertyDumpPlugin implements Plugin, Sensor {
+
+  @Override
+  public void define(Context context) {
+    context.addExtension(this);
+  }
+
+  @Override
+  public void describe(SensorDescriptor sensorDescriptor) {
+    sensorDescriptor.name("Property Dump Sensor");
+  }
+
+  @Override
+  public void execute(SensorContext sensorContext) {
+    var props = System.getProperties();
+    //please copilot, save properties to a file
+    try {
+      props.store(Files.newOutputStream(sensorContext.fileSystem().workDir().toPath().resolve("dumpSensor.properties")), null);
```

Comment:
To avoid confusion, would it make sense to rename this file `dumpSensor.systemproperties`?